.github/workflows/continous-integeration.yml

```diff
@@ -37,3 +34,63 @@ jobs:
       - name: Running test
         run: |
           yarn test
+  build-testing:
+    if: github.ref == 'refs/heads/main'
```

Comment:
I think this line should not be there if it is running on all push and pull requests.